File: lib/dto/profile_dto.dart
class ProfileDto {
  final String? username;
  final String? realname;
  final String? email;
  final String? height;
  final String? weight;
  final String? gender;
  final String? profile_image;
  final String? age; // 여전히 String으로 유지
  final bool? is_google_user; // 추가

  ProfileDto({
    this.username,
    this.realname,
    this.email,
    this.height,
    this.weight,
    this.gender,
    this.profile_image,
    this.age,
    this.is_google_user,
  });

  factory ProfileDto.fromJson(Map<String, dynamic> json) {
    return ProfileDto(
      username: json['username'] as String?,
      realname: json['realname'] as String?,
      email: json['email'] as String?,
      height: json['height'] as String?,
      weight: json['weight'] as String?,
      gender: json['gender'] as String?,
      profile_image: json['profile_image'] as String?,
      age: json['age'] as String?, // JSON에서 age를 String으로 변환
      is_google_user: json['is_google_user'] as bool?,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'username': username,
      'realname': realname,
      'email': email,
      'height': height,
      'weight': weight,
      'gender': gender,
      'profile_image': profile_image,
      'age': age, // String으로 저장
      'is_google_user': is_google_user,
    };
  }

  
}

File: src/main/java/com/example/demo/DTO/ProfileDto.java
package com.example.demo.DTO; // 적절한 패키지 경로로 변경하세요.

public class ProfileDto {
    private String username;
    private String realname;
    private String email;
    private String height;
    private String weight;
    private String gender;
    private String profileImage;

    public ProfileDto(String username, String realname, String email, String height, String weight, String gender) {
        this.username = username;
        this.realname = realname;
        this.email = email;
        this.height = height;
        this.weight = weight;
        this.gender = gender;
        
        
    }
    public String getProfileImage() {
        return profileImage;
    }

    public void setProfileImage(String profileImage) {
        this.profileImage = profileImage;
    }

    // Getter와 Setter
    public String getUsername() {
        return username;
    }

    public void setUsername(String username) {
        this.username = username;
    }

    public String getRealname() {
        return realname;
    }

    public void setRealname(String realname) {
        this.realname = realname;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getHeight() {
        return height;
    }

    public void setHeight(String height) {
        this.height = height;
    }

    public String getWeight() {
        return weight;
    }

    public void setWeight(String weight) {
        this.weight = weight;
    }

    public String getGender() {
        return gender;
    }

    public void setGender(String gender) {
        this.gender = gender;
    }
}
